# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1226660675 -3600
#      Fri Nov 14 12:04:35 2008 +0100
# Node ID 655193554ad21bc5b45f49dc28bb105fb1380d26
# Parent  f32a91d991561ca2cbdee5be8842613b7f56b2d8
Correct use of resize_fill_value in Array2<T>::index methods.

diff --git a/liboctave/Array2.h b/liboctave/Array2.h
--- a/liboctave/Array2.h
+++ b/liboctave/Array2.h
@@ -112,24 +112,24 @@ public:
 
   Array2<T> hermitian (T (*fcn) (const T&) = 0) const
     {
       Array<T> tmp = Array<T>::hermitian (fcn);
       return Array2<T> (tmp, tmp.rows (), tmp.columns ());
     }
 
   Array2<T> index (idx_vector& i, int resize_ok = 0,
-		   const T& rfv = resize_fill_value (T ())) const
+		   const T& rfv = Array<T>::resize_fill_value ()) const
     {
       Array<T> tmp = Array<T>::index (i, resize_ok, rfv);
       return Array2<T> (tmp, tmp.rows (), tmp.columns ());
     }
 
   Array2<T> index (idx_vector& i, idx_vector& j, int resize_ok = 0,
-		   const T& rfv = resize_fill_value (T ())) const
+		   const T& rfv = Array<T>::resize_fill_value ()) const
     {
       Array<T> tmp = Array<T>::index (i, j, resize_ok, rfv);
       return Array2<T> (tmp, tmp.rows (), tmp.columns ());
     }
 
   Array2<T> sort (octave_idx_type dim = 0, sortmode mode = ASCENDING) const
     {
       Array<T> tmp = Array<T>::sort (dim, mode);
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -12,16 +12,21 @@ 2008-11-12  Michael Goffioul  <michael.g
 	* oct-inttypes.cc (octave_int<T>::type_name): Ditto.
 	* oct-inttypes.cc (powf(const octave_int<T>&,const float&)): Cast 'b'
 	to double.
 	* oct-inttypes.h: Undefine min/max.
 	* oct-norm.h (xnorm, xfrobnorm, xcolnorms, xrownorms): Add OCTAVE_API
 	tag.
 	* oct-norm.cc (xnorm, xfrobnorm, xcolnorms, xrownorms): Ditto.
 
+2008-11-14  David Bateman  <dbateman@free.fr>
+
+	* Array2.h (Array2<T> Array2<T>::index): Correct use of
+	resize_fill_value.
+
 2008-11-12  Jaroslav Hajek <highegg@gmail.com>
 
 	* DiagArray2.h (DiagArray2<T>::maybe_delete_elements): Remove
 	declaration.
 
 2008-11-09  Jaroslav Hajek <highegg@gmail.com>
 
 	* oct-norm.cc: Don't include Array.cc.  
